### PR TITLE
CDAP-15594 batch system metadata

### PR DESCRIPTION
**Pull request by yaojiefeng**
JIRA: https://issues.cask.co/browse/CDAP-15594
build: https://builds.cask.co/browse/CDAP-DUT6995-1

The system metadata writer writes the app and program metadata one by one. The call should be batched to reduce the RPC cost.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/system/AbstractSystemMetadataWriter.java [hunk 1]
```diff
@@ -60,15 +60,18 @@
    */
   @Override
   public void write() {
```
Comment (anew) [inline cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/system/AbstractSystemMetadataWriter.java:62]: is the write() method still needed?

File: cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/writer/NoOpMetadataServiceClient.java [hunk 2]
```diff
@@ -42,4 +44,9 @@ public void update(MetadataMutation.Update updateMutation) {
   public void remove(MetadataMutation.Remove removeMutation) {
     // No Op
   }
+
+  @Override
+  public void batch(List<MetadataMutation> mutations) {
+
```
Comment (anew) [inline cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/writer/NoOpMetadataServiceClient.java]: empty body should have a "// no-op" comment. I am surprised checkstyle allows this.

**Review by anew — APPROVED** [review]
Only minor comments.

File: cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/system/AbstractSystemMetadataWriter.java (hunk 1 above)
Comment (yaojiefeng) [reply cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/system/AbstractSystemMetadataWriter.java:62]: Yes it is still used by the artifact and dataset to write system metadata

**Pull request merged**